mono/utils/mono-threads-coop.c

```diff
@@ -67,10 +77,36 @@ mono_threads_state_poll (void)
 	}
 }
 
+static void *
+return_stack_ptr ()
+{
+	int i;
+	return &i;
+}
+
+static void
+copy_stack_data (MonoThreadInfo *info, void* stackdata_begin, void* stackdata_end)
+{
+	MonoThreadUnwindState *state;
+	int stackdata_size;
+
+	state = &info->thread_saved_state [SELF_SUSPEND_STATE_INDEX];
+
+	stackdata_size = (char*)stackdata_begin - (char*)stackdata_end;
+
+	state->gc_stackdata = mono_mempool_alloc (info->stackdata_mp, stackdata_size);
```

Comment:
IIUC our mempool is just an abstraction over a simple linear memory allocator model which is exactly what we need in this instance. Why would you have me basically duplicate its code instead of re-using the abstractions we already have?